### PR TITLE
Feat: Added datetime widget for adjusting date and time formats

### PR DESCRIPTION
**Pull request by AustinCGause**
## Minor changes

- Adds a config option that allows for the selection of either 12H or 24H time formats

## Additional informations

Keeps the default time format of 24H, but allows users to update their configuration to be 12H time format with AM/PM identifier.

### CODE REVIEW

File: src/time-format.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+# Read the preferred time format from the argument
+time_format="$1"
+
+if [[ "$time_format" == "12H" ]]; then
+    # 12-hour format with AM/PM
+    echo "%I:%M %p"
```
Comment (janoamaral) [inline src/time-format.sh:8]: show date and time `#[fg=#a9b1d6,bg=#24283B] %Y-%m-%d #[]❬ %I:%M %p`

File: src/time-format.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+else
+    # Default 24-hour format
+    echo "%H:%M"
```
Comment (janoamaral) [inline src/time-format.sh:11]: show date and time `#[fg=#a9b1d6,bg=#24283B] %Y-%m-%d #[]❬ %H:%M`

File: tokyo-night.tmux [hunk 3]
```diff
@@ -58,5 +62,5 @@ tmux set -g window-status-current-format "#[fg=#44dfaf,bg=#1F2335]   #[fg=#a9
 tmux set -g window-status-format "#[fg=#c0caf5,bg=default,none,dim]   $window_number #W#[nobold,dim]#{?window_zoomed_flag, $zoom_number, $custom_pane}#[fg=yellow,blink] #{?window_last_flag,󰁯 ,} "
 
 #+--- Bars RIGHT ---+
-tmux set -g status-right "$cmus_status#[fg=#a9b1d6,bg=#24283B]$netspeed$git_status$wb_git_status#[fg=#a9b1d6,bg=#24283B]  %Y-%m-%d #[]❬ %H:%M "
+tmux set -g status-right "$cmus_status#[fg=#a9b1d6,bg=#24283B]$netspeed$git_status$wb_git_status#[fg=#a9b1d6,bg=#24283B]  %Y-%m-%d #[]❬ $time_format_string "
```
Comment (janoamaral) [inline tokyo-night.tmux:65]: replace `[fg=#a9b1d6,bg=#24283B] %Y-%m-%d #[]❬ $time_format_string ` with `$time_widget`

**Review by janoamaral — CHANGES_REQUESTED** [review]
First, thanks for your contribution! I believe that we should make time indicator an independent widget. You are almost there in fact. If you move date and rename `time-format.sh` to `time-widget.sh` I'll merge it

**Comment by AustinCGause** [comment]
I have adjusted the date and time formatting to be a separate widget and have updated the README to reflect the new changes. Feel free to let me know if I need to adjust anything else. I have tested the changes and they work correctly.

**Comment by AustinCGause** [comment]
I apologize, I missed the comment about the option for enabling and disabling, I will implement it.

**Comment by AustinCGause** [comment]
I have tested the new logic as well, and everything is working as intended, please let me know if there are any adjustments that need to be made.

File: tokyo-night.tmux [hunk 4]
```diff
@@ -58,5 +59,5 @@ tmux set -g window-status-current-format "#[fg=#44dfaf,bg=#1F2335]   #[fg=#a9
 tmux set -g window-status-format "#[fg=#c0caf5,bg=default,none,dim]   $window_number #W#[nobold,dim]#{?window_zoomed_flag, $zoom_number, $custom_pane}#[fg=yellow,blink] #{?window_last_flag,󰁯 ,} "
 
 #+--- Bars RIGHT ---+
-tmux set -g status-right "$cmus_status#[fg=#a9b1d6,bg=#24283B]$netspeed$git_status$wb_git_status#[fg=#a9b1d6,bg=#24283B]  %Y-%m-%d #[]❬ %H:%M "
+tmux set -g status-right "$cmus_status#[fg=#a9b1d6,bg=#24283B]$netspeed$git_status$wb_git_status$date_and_time "
```
Comment (janoamaral) [inline tokyo-night.tmux:62]: remove extra space after `$date_and_time`

File: src/datetime-widget.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+if [[ "$time_format" == "12H" ]]; then
+    # 12-hour format with AM/PM
+    time_string="%I:%M %p"
+else
+    # Default to 24-hour format if not specified
+    time_string="%H:%M"
+fi
+
+echo "#[fg=#a9b1d6,bg=#24283B] $date_string #[]❬ $time_string"
```
Comment (janoamaral) [inline src/datetime-widget.sh:38]: add an extra space after `$time_string`

**Review by janoamaral — CHANGES_REQUESTED** [review]
Fix those 2 comments and we are good to go

**Comment by AustinCGause** [comment]
I have resolved your comments, but also, sorry if this is a bit of a noob question, but what does "resolve conversation" mean/do on the comment's that you added? Is it standard practice to click that after fixing what the comments refer to? Just curious for future reference.

**Review by janoamaral — APPROVED** [review]

**Pull request merged**